backend/proxy_server/auth.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+)
+
+// global oauth providers' client config
+var (
+	githubConfig *oauth2.Config
+)
+
+// providers' client parameters
+const (
+	GithubClientID          = "4e5058f5e68e11b91193"
+	GithubClientSecret      = "188cbe058b6bfeeb856f4962806e72342e2da3d9"
+	GithubClientRedirectURL = "http://106.14.177.57/login"
```

Comment:
I have got a question here🤔. 
Where is the URL from? Is it a github server or selfhosted server? Does this need to be updated regularly?

Reply:
It's a self-hosted server and here is my server IP.  You need to set this in your github account, then you will get `clientID`,`secret`. Maybe you can find it at `https://github.com/settings/developers`. For more details: `https://docs.github.com/cn/developers/apps/building-oauth-apps/creating-an-oauth-app`. As well, we need to set ClientID in front-end page.